File: 3038-maximum-number-of-operations-with-the-same-score-i/3038-maximum-number-of-operations-with-the-same-score-i.cpp
class Solution {
public:
    int maxOperations(vector<int>& nums) {
        vector<pair<int, int>> vec;
        for (int i = 0; i < nums.size(); i += 2) {
            if (i < nums.size() - 1) { 
                vec.push_back({nums[i], nums[i + 1]});
            }
        }
        vector<int> arr;
        for (auto it : vec) {
            arr.push_back(it.first + it.second);
        }
        int ans = 1;
        for (int i = 1; i < arr.size(); i++) {
            if (arr[i] == arr[0]) {
                ans++;
            } else {
                break;
            }
        }
        return ans;
    }
};

File: 3038-maximum-number-of-operations-with-the-same-score-i/3038-maximum-number-of-operations-with-the-same-score-i.java
class Solution {
    public int maxOperations(int[] nums) {
        if(nums.length < 2) return 0;
        if(nums.length == 2) return 1;
        
        int count = 1;
        int score = nums[0] + nums[1];
        int i = 2;
        
        while(i < nums.length) {
            if(i+1 < nums.length && nums[i] + nums[i+1] == score) {
                count++;
                i += 2;
            } else {
                break;
            }
            
        }
        
        return count;
    }
}